File: TSP/code/_electoralMethods/electoralsystems/util/HighestPriorityStrategy.java
package _electoralMethods.electoralsystems.util;

public class HighestPriorityStrategy implements WinnerStrategy {
	@Override
	public int winner(int minPriorityIndex, int maxPriorityIndex) {
		return maxPriorityIndex;
	}

	@Override
	public int looser(int minPriorityIndex, int maxPriorityIndex) {
		return minPriorityIndex;
	}

	@Override
	public boolean leftIsBetter(double a, double b) {
		return a > b;
	}

	@Override
	public WinnerStrategy getInverse() {
		return new LowestPriorityStrategy();
	}
}

File: crp/crp/HighestPriorityStrategy.cpp
#include "HighestPriorityStrategy.h"

#include <memory>

#include "LowestPriorityStrategy.h"

int HighestPriorityStrategy::winner(int minPriorityIndex, int maxPriorityIndex)
{
	return maxPriorityIndex;
}

int HighestPriorityStrategy::looser(int minPriorityIndex, int maxPriorityIndex)
{
	return minPriorityIndex;
}

bool HighestPriorityStrategy::leftIsBetter(double a, double b)
{
	return a > b;
}

std::shared_ptr<WinnerStrategy> HighestPriorityStrategy::getInverse()
{
	return std::make_shared<LowestPriorityStrategy>();
}
